### ISSUE TITLE
Terminating expressions fail when function lifted to measures

### ISSUE THREAD

**Issue by nikivazou** [ucsd-progsys/liquidhaskell#1245]
e.g., 

```
import qualified Data.Set as S 


{-@ data Map [size] @-}
data Map k v =
    Empty
  | Bind k v (Map k v)

{-@ measure size @-}
{-@ size :: Map k v -> Nat @-}
size :: Map k v -> Int
size Empty = 0
size (Bind _ _ r) = 1 + size r

{-@ measure keys @-}
{-@ keys :: zzz: Map k v -> S.Set k / [size zzz, 0] @-}
keys :: Ord k => Map k v -> S.Set k
keys Empty = S.empty
keys (Bind k _ r) = add k r

{-@ inline add @-}
{-@ add :: k -> zzz:Map k v -> S.Set k / [size zzz, 1] @-}
add :: Ord k => k -> Map k v -> S.Set k
add k r = S.singleton k `S.union` keys r
```

**Comment by ranjitjhala** [comment]
Wow, thats _super_ surprising (that we didn't have a single regression test
for this case?!)

**Comment by ranjitjhala** [comment]
Simplified a bit:

```haskell
{-@ data Map [size] @-}
data Map k v =
    Empty

{-@ measure size @-}
{-@ size :: Map k v -> Nat @-}
size :: Map k v -> Int
size Empty = 0

{-@ measure keys @-}
{-@ keys :: zzz:_ -> Int / [size zzz] @-}
keys :: Map k v -> Int
keys Empty = 12
```

**Comment by nikivazou** [comment]
I think I fixed it... 

Not totally surprised, when I saw this comment :) 

https://github.com/ucsd-progsys/liquidhaskell/blob/develop/src/Language/Haskell/Liquid/Bare.hs#L867

**Comment by ranjitjhala** [comment]
Also, 

https://github.com/ucsd-progsys/liquidhaskell/blob/develop/src/Language/Haskell/Liquid/Bare.hs#L917

:)

**Comment by nikivazou** [comment]
I saw it. Incomprehensible, so ignore :) 

This was preventing this bug, but now things have changed

**Comment by nikivazou** [comment]
In short, when one has to write comments, something is going to break :)

**Comment by nikivazou** [comment]
Fixed with #1246

**Comment by ranjitjhala** [comment]
Yay!
